portal/worker/src/caching.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { resolveAndFetchPage } from "@lib/page_fetching";
+import { SuiClient, getFullnodeUrl } from "@mysten/sui/client";
+import { NETWORK } from "@lib/constants";
+import { DomainDetails } from "@lib/types";
+
+const CACHE_NAME = "walrus-sites-cache";
+// TODO - move it to .env
+const CACHE_EXPIRATION_TIME = 24 * 60 * 60 * 1000 // 24 hours in milliseconds
+
+/**
+* Respond to the request using the cache API.
+*/
+export default async function respondUsingCache(
+    event: FetchEvent, parsedUrl: DomainDetails, urlString: string
+) {
```

Comment:
Suggestions: with the changes above, the signature can be simplified to
```suggestion
export default async function resolveWithCache(parsedUrl: DomainDetails): Promise<Response> {
```
and the internal `event.respondWith` can be removed

Reply:
Fair 💯